CMakeLists.txt

```diff
@@ -16,7 +16,16 @@ set(VM_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src/vm)
 
 if(CMAKE_SYSTEM_NAME STREQUAL Linux)
     set(CLR_CMAKE_PLATFORM_UNIX 1)
-    set(CLR_CMAKE_PLATFORM_UNIX_TARGET_AMD64 1)
+    if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64)
+        set(CLR_CMAKE_PLATFORM_UNIX_TARGET_AMD64 1)
+    elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l)
```

Comment:
armv7l is little-endian armv7, but you seem to be targeting hard-float ABI, which is (generally) known as armv7hl.  Perhaps this isn't consistent across kernels/cmakes?  Can you add armv7hl as well?